org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/util/JavaProjectClasspathChangeAnalyzer.java

```diff
@@ -84,6 +102,52 @@ public boolean isAttachmentChangeOnly(IJavaElementDelta delta) {
 		return true;
 	}
 	
+	private static Version installedJdtCoreVersion;
+	
+	private static boolean isJdtCoreGreaterOrEqual(Version version) {
+		if (installedJdtCoreVersion == null) {
+			installedJdtCoreVersion = JavaCore.getPlugin().getBundle().getVersion();
+		}
+		return installedJdtCoreVersion.compareTo(version) >= 0;
+	}
+
+	/**
+	 * this method checks if the change is a source or javadoc (de)attachment change only.
+	 * @since 2.30
+	 */
+	@Beta
+	protected boolean isAttachmentChangeFlagOnly(IJavaElementDelta child) {
+		int flags = child.getFlags();
+		if (flags == 0) {
+			return false;
+		}
+		int flagsWithoutSourceAttachments = flags & ~(IJavaElementDelta.F_SOURCEATTACHED | IJavaElementDelta.F_SOURCEDETACHED);
+		if (flagsWithoutSourceAttachments == 0) {
+			// flags were attached/detached only
+			return true;
+		}
+		if (flagsWithoutSourceAttachments == 0x800000) { // TODO use IJavaElementDelta.F_CLASSPATH_ATTRIBUTES
```

Comment:
This mask should be checked against the original flags:
```suggestion
		if ((flags & 0x800000) != 0) { // TODO use IJavaElementDelta.F_CLASSPATH_ATTRIBUTES
```

Reply:
but what if the source attachment AND the jar changes.
then i dont want to go into the if and directly report "classpath change"

Reply:
(see also my review comment on the "old" code

Reply:
and of course my jdt knowledge is too small to create a test for that

Reply:
Ah, yes. Of course.